mantidproject/mantid
# New intercept type determination to resolve negative L12 observed in multiple scattering correction

**Description of work.**
- Original Gitlab user story: [PD380](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/380)
- This is #32966 into `main`

The original method to determine the type of the intercept between a line and cylinder is by twiddling a fixed step size along the line direction, which is sensitive to the input geometry.
Instead of tweaking the fixed step size, this PR uses the mid point between the intercept of interest and the one before it to determine the type of the intercept.
This change also helps resolve the issue where negative L12_container is found when certain hollow cylinder geometry is used in MultipleScatteringCorrection.
Also, the math behind the determination of the intercept between a line and a cylinder is now documented in `Concept::Geometry of Shape`.



**To test:**
This is a Quality of Life (QoL) change, so no explicit testing script is needed.
However, one can use the following unit test to confirm the effect of the changes
- MultipleScattreingCorrectionTest
- RasterizeTest
- LineTest










---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
